volmdlr/faces.py

```diff
@@ -6975,7 +6975,11 @@ def to_step(self, current_id):
         step_content = ''
         face_ids = []
         for face in self.faces:
-            face_content, face_sub_ids = face.to_step(current_id)
+            if isinstance(face, Face3D):
+                face_content, face_sub_ids = face.to_step(current_id)
+            else:
```

Comment:
Why it is different when the face is not a Face3D ?

Reply:
Face3D returns a list of ids, the ClosedShell3D returned only one id. I have changed all to_step methods to return a list of ids. I've deleted this verification

Reply:
When it's a face, face_sub_ids is a liste of integers. But then it is a closedShell3D, it is a single integer. We have this weird behaviour where a closedshell3d is passed as Face in faces...